CMakeLists.txt

```diff
@@ -0,0 +1,126 @@
+CMAKE_MINIMUM_REQUIRED(VERSION 3.10 FATAL_ERROR)
+
+# Set a default build type if none was specified
+SET(default_build_type "Release")
+IF(EXISTS "${CMAKE_SOURCE_DIR}/.git")
+    SET(default_build_type "Debug")
+ENDIF()
+
+IF(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+    MESSAGE(STATUS "Setting build type to '${default_build_type}' as none was specified.")
+    SET(CMAKE_BUILD_TYPE ${default_build_type} CACHE STRING "Choose the type of build." FORCE)
+    # Set the possible values of build type for cmake-gui
+    SET_PROPERTY(
+        CACHE CMAKE_BUILD_TYPE
+        PROPERTY
+            STRINGS
+            "Debug"
+            "Release"
+            "MinSizeRel"
+            "RelWithDebInfo"
+    )
+ENDIF()
+
+PROJECT(NUbotsGazebo VERSION 1.0.0)
+
+# Find Gazebo and Ignition Transport
+FIND_PACKAGE(Protobuf REQUIRED)
+
+# Build the custom protobuf messages Build our builtin protobuf classes
+FILE(GLOB_RECURSE protobufs "shared/**.proto")
+FOREACH(proto ${protobufs})
+    # Get the file without the extension and the directory
+    FILE(
+        RELATIVE_PATH
+        proto
+        ${CMAKE_CURRENT_SOURCE_DIR}/shared
+        ${proto}
+    )
+    GET_FILENAME_COMPONENT(file_we ${proto} NAME_WE)
+    GET_FILENAME_COMPONENT(dir ${proto} DIRECTORY)
+
+    # Run the protocol buffer compiler on these new protobufs
+    ADD_CUSTOM_COMMAND(
+        OUTPUT
+            "${CMAKE_CURRENT_BINARY_DIR}/shared/${dir}/${file_we}.pb.cc"
+            "${CMAKE_CURRENT_BINARY_DIR}/shared/${dir}/${file_we}.pb.h"
+        MAIN_DEPENDENCY shared/${proto}
+        COMMAND
+            ${PROTOBUF_PROTOC_EXECUTABLE}
+            ARGS
+            --cpp_out=lite:${CMAKE_CURRENT_BINARY_DIR}/shared
+            -I${CMAKE_CURRENT_SOURCE_DIR}/shared
+            "${CMAKE_CURRENT_SOURCE_DIR}/shared/${dir}/${file_we}.proto"
+        COMMENT "Compiling protocol buffer ${proto}"
+    )
+
+    LIST(
+        APPEND
+            msg_src "${CMAKE_CURRENT_BINARY_DIR}/shared/${dir}/${file_we}.pb.cc"
+            "${CMAKE_CURRENT_BINARY_DIR}/shared/${dir}/${file_we}.pb.h"
+    )
+ENDFOREACH(proto)
+
+IF(NOT MSVC)
+    # Compilation must be done with c++14 for NUClear to work
+    ADD_COMPILE_OPTIONS(-std=c++14 -fPIC)
```

Comment:
We don't want -fPIC it's already done on line 78. And std c++14 should be done as per line 79